packages/lib-classifier/src/store/WorkflowStore/Workflow/Workflow.js

```diff
@@ -73,6 +73,14 @@ const Workflow = types
       return activeSet?.id
     },
 
+    get hasSurveyTask() {
+      const anySurveyTasks = self.tasks && Object.values(self.tasks).some(task => {
+        return task.type === 'survey'
+      })
+
+      return anySurveyTasks
+    },
```

Comment:
If I can make a code design suggestion here, I would add a check for `self.hasSurveyTask` to `workflow.layout`, then return a new layout for survey tasks. The advantage of that would be that you avoid accidentally introducing side effects into layouts that are already in use for other projects. For example, that would avoid accidental changes to transcription workflows, when all you want to do is change the layout for camera survey workflows.

That change would also mean that you don't change the API design for layouts in general. At the moment, each existing layout has to handle a new `hasSurveyTask` argument.

Reply:
Agreed! That would make much more sense to create a SurveyTask layout component rather than impact the existing ones.

Reply:
Although what would happen if a workflow has a survey task, but also `configuration.limit_subject_height` set as `true` 🤔

Reply:
Oh, that’s a good point. Layout settings can’t be easily combined.

Reply:
A couple of ideas:

1. use a boolean attribute/prop to limit the subject height:
```jsx
<SurveyTaskLayout limitSubjectHeight />
```
2. use composition to limit the subject height:
```jsx
<LimitedHeightLayout>
  <SurveyTaskLayout />
</LimitedHeightLayout>
```

Reply:
A workflow with a survey task could use the default `MaxWidth`, subject height limited `CenteredLayout`, or if the workflow tasks also contain a transcription task the `NoMaxWidth` layout.

Reply:
The limited height setting causes some weird layout bugs too.
- https://github.com/zooniverse/front-end-monorepo/issues/6275
- https://github.com/zooniverse/front-end-monorepo/issues/6061